source/MRViewer/MRPalette.cpp

```diff
@@ -511,13 +511,27 @@ float Palette::getRelativePos( float val ) const
         return ( val - parameters_.ranges[0] ) / ( parameters_.ranges[1] - parameters_.ranges[0] );
     else if ( parameters_.ranges.size() == 4 )
     {
-        if ( val >= parameters_.ranges[1] && val <= parameters_.ranges[2] )
-            return 0.5f;
+        float centralZoneSize = parameters_.ranges[2] - parameters_.ranges[1];
+        bool isInCentralZone = val >= parameters_.ranges[1] && val <= parameters_.ranges[2];
+        if ( isInCentralZone && ( texture_.filter == FilterType::Linear || centralZoneSize <= 0.0f ) )
+                return 0.5f;
+
+        float uvRange = 0.5f;
+        float uvCenterMax = 0.5f;
+        if ( texture_.filter == FilterType::Discrete )
+        {
+            auto realDiscretization = ( 2 * parameters_.discretization + 1 );
+            uvRange = float( parameters_.discretization ) / realDiscretization;
+            uvCenterMax = float( parameters_.discretization + 1 ) / realDiscretization;
```

Comment:
not clear names (function does not work with uv, it scales from [nMax, nMin, pMin, pMax] to [0, 1])
mb better centralZoneMin & centralZoneMax ?